cloud/amazon/ec2_vpc.py

```diff
@@ -227,6 +227,81 @@ def find_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
     return (found_vpc)
 
+#Check if the route table has all routes as in given list
+def routes_match(rt_list=None, rt=None, igw=None):
+    local_routes = []
+    remote_routes = []
+    for route in rt_list:
+        route_kwargs = {}
+        if route['gw'] == 'igw':
+            route_kwargs['gateway_id'] = igw.id
+            route_kwargs['instance_id'] = None
+            route_kwargs['state'] = 'active'
+        elif route['gw'].startswith('i-'):
+            route_kwargs['instance_id'] = route['gw']
+            route_kwargs['gateway_id'] = None
+            route_kwargs['state'] = 'active'
+        else:
+            route_kwargs['gateway_id'] = route['gw']
+            route_kwargs['instance_id'] = None
+            route_kwargs['state'] = 'active'
+        route_kwargs['destination_cidr_block'] = route['dest']
+        local_routes.append(route_kwargs)
+    for j in rt.routes:
+        remote_routes.append(j.__dict__)
+    match = []
+    for i in local_routes:
+        change = "false"
+        for j in remote_routes:
+            if set(i.items()).issubset(set(j.items())):
+                change = "true"
+        match.append(change)
+    if 'false' in match:
+        return False
+    else:
+        return True
+
+#Check if the remote routes match the local routes
+def rtb_match(route_tables=None, vpc_conn=None, module=None, vpc=None, igw=None):
+    needs_change = False
+    #Check if we have any other extra route tables
+    #We add a one for the main table
+    rtb_len = len(route_tables) + 1
+    remote_rtb_len = len(vpc_conn.get_all_route_tables(filters={'vpc_id': vpc.id}))
+    if remote_rtb_len != rtb_len:
+        needs_change = True
+    for rt in route_tables:
+        if needs_change:
+            break;
```

Comment:
Instead of breaking out of the loop, you could change line 272 to `return True`.

The same thing can be done for line 278, 288, 300 and 301.

If you do that you can also remove line 266 and change 302 to `return False`